src/ota_proxy/server_app.py

```diff
@@ -52,20 +52,20 @@
 # only expose app
 __all__ = ("App",)
 
-
 # helper methods
 
 
-def parse_raw_headers(raw_headers: List[Tuple[bytes, bytes]]) -> Dict[str, str]:
+def parse_raw_headers(raw_headers: list[tuple[bytes, bytes]]) -> CIMultiDict[str]:
```

Comment:
Use case-insensitive dict from the very beginning of parsing headers to prevent accidentally dropping any headers.